File: 2025-08/2025-08-16/anagram_checker.py
def are_anagrams(str1, str2):
    import re

    regex1 = "[^" + "".join(set(re.compile(r"[^\s]").findall(str1.lower()))) + "]"
    regex2 = "[^" + "".join(set(re.compile(r"[^\s]").findall(str2.lower()))) + "]"

    return len(str1) == len(str2) \
        and not re.compile(regex1, re.I).search(str2.replace(" ", "")) \
        and not re.compile(regex2, re.I).search(str1.replace(" ", ""))

print(are_anagrams("listen", "silent"))
print(are_anagrams("School master", "The classroom"))
print(are_anagrams("A gentleman", "Elegant man"))
print(are_anagrams("Hello", "World"))
print(are_anagrams("apple", "banana"))
print(are_anagrams("cat", "dog"))

print(are_anagrams("abc", "abb"))
print(are_anagrams("abc", "abbc"))

# --- TEST SUITE ---

tests_text = '''
1. are_anagrams("listen", "silent") should return True.
2. are_anagrams("School master", "The classroom") should return True.
3. are_anagrams("A gentleman", "Elegant man") should return True.
4. are_anagrams("Hello", "World") should return False.
5. are_anagrams("apple", "banana") should return False.
6. are_anagrams("cat", "dog") should return False.
'''

import re

tests_regex = re.compile(r"(?P<number>\d+)\.\s(?P<function_call>.+) should return (?P<output>.+)\.$", re.MULTILINE)
test_data = [item.groupdict() for item in re.finditer(tests_regex, tests_text)]

def run_tests(test_data):
    print("--------------------------")
    print("🧪 Starting Verification...")
    print("--------------------------")

    all_passed = True
    fail_count = 0

    for test in test_data:
        function_call_output = eval(test['function_call'])
        test_output = eval(test['output'])
        
        if function_call_output == test_output:
            print(f"{test['number']}.✅ PASS - Function Call:\n{test['function_call']}")
        else:
            print(f"{test['number']}.❌ FAIL - Function Call:\n{test['function_call']}\nExpected:\n{test_output}\nGot:\n{function_call_output}")
            all_passed = False
            fail_count += 1

    print("----------------------------");

    if (all_passed):
        print("🎉 SUCCESS: All tests PASSED.");
    else:
        print(f"⚠️ WARNING: {fail_count}/{len(test_data)} tests FAILED.");

run_tests(test_data)

File: 2025-08/2025-08-16/anagramChecker.js
function areAnagrams(str1, str2) {
  const regex1 = "[^" + [...new Set(str1.toLowerCase().match(/[^\s]/g))].join("") + "]";
  const regex2 = "[^" + [...new Set(str2.toLowerCase().match(/[^\s]/g))].join("") + "]";
  
  return str1.length === str2.length
    && !new RegExp(regex1, "i").exec(str2.replace(" ", ""))
    && !new RegExp(regex2, "i").exec(str1.replace(" ", ""));
}

console.log(areAnagrams("listen", "silent"));
console.log(areAnagrams("School master", "The classroom"));
console.log(areAnagrams("A gentleman", "Elegant man"));
console.log(areAnagrams("Hello", "World"));
console.log(areAnagrams("apple", "banana"));
console.log(areAnagrams("cat", "dog"));

console.log(areAnagrams("abc", "abb"));
console.log(areAnagrams("abc", "abbc"));

// --- TEST SUITE ---

const testsText = `
1. areAnagrams("listen", "silent") should return true.
2. areAnagrams("School master", "The classroom") should return true.
3. areAnagrams("A gentleman", "Elegant man") should return true.
4. areAnagrams("Hello", "World") should return false.
5. areAnagrams("apple", "banana") should return false.
6. areAnagrams("cat", "dog") should return false.
`;

const testsRegex = /(?<number>\d+)\.\s(?<functionCall>.+) should return (?<output>.+)\.$/gm;
const testData = [...testsText.matchAll(testsRegex).map(match => match.groups)];

function runTests(testData) {
  console.log("--------------------------");
  console.log("🧪Starting Verification...");
  console.log("--------------------------");

  let allPassed = true;
  let failCount = 0;
  
  testData.forEach(test => {
    const functionCallOutput = eval(test.functionCall);
    const testOutput = eval(test.output);
    let comparison;

    if (Array.isArray(testOutput))
      comparison = arraysEqual(functionCallOutput, testOutput);
    else
      comparison = functionCallOutput === testOutput;

    if (comparison) {
      console.log(`${test.number}.✅PASS - Function Call: ${test.functionCall}`);
    }
    else {
      console.log(`${test.number}.❌FAIL - Function Call: ${test.functionCall}\nExpected: ${testOutput}\nGot: ${functionCallOutput}`);
      allPassed = false;
      failCount ++;
    }
  })

  console.log("----------------------------");

  if (allPassed)
    console.log("🎉SUCCESS: All tests PASSED.");
  else
    console.log(`⚠️WARNING: ${failCount}/${testData.length} tests FAILED.`);
}

function arraysEqual(a, b) {
  if (a.length !== b.length)
    return false;

  for (const index in a) {
    if (Array.isArray(a[index])) {
      if (Array.isArray(b[index])) {
        if (!arraysEqual(a[index], b[index]))
          return false;
      }
      else 
        return false;
    }

    else if (a[index] !== b[index])
      return false;
  }

  return true;
}

runTests(testData);